code/game/machinery/atmoalter/canister.dm

```diff
@@ -400,54 +411,107 @@ update_flag
 			release_pressure = min(10*ONE_ATMOSPHERE, release_pressure+diff)
 		else
 			release_pressure = max(ONE_ATMOSPHERE/10, release_pressure+diff)
-
-	if (href_list["relabel"])
+		
+	if (href_list["rename"])
 		if (can_label)
+			var/T = input("Choose canister label", "Name", name)
+			if (can_label) //Exploit prevention
+				if (T)
+					name = T
+				else
+					name = "Canister"
+			else
+				usr << "\red As you attempted to rename it the pressure rose!"
 
-			var/label1 = input("Choose canister label", "Primary color") as null|anything in possiblemaincolor
-
-			var/label2 = input("Choose canister label", "Secondary color") as null|anything in possibleseccolor
-
-			var/label3 = input("Choose canister label", "Tertiary color") as null|anything in possibletertcolor
-
-			var/label4 = input("Choose canister label", "Quaternary color") as null|anything in possiblequartcolor
-
-			decals = list()
-
-			_color = list(
-			(label1 ? possiblemaincolor[label1] : color[1]),//if the user didn't specify a primary colour, keep the current one.
-			possibleseccolor[label2],
-			possibletertcolor[label3],
-			possiblequartcolor[label4]
-			)
-
-			decals = list()
-
-			var/list/tempposdecals = list()
-
-			for(var/d in possibledecals)//populate the temp list
-				tempposdecals.Add(d)
-				tempposdecals[d] = possibledecals[d]
-
-			while (src && !src.gc_destroyed && usr)//allow the user to select (theoretically) INFINITE DECALS!!!
-				var/newdecal = input("Choose canister label", "Decal") as anything in tempposdecals
-				if (newdecal == "Done")
-					break
-				decals.Add(tempposdecals[newdecal])
-				tempposdecals.Remove(newdecal)
-
-			src.name = (input("Choose canister label", "Name") as text) + " canister"
-
+	if (href_list["choice"] == "Primary color")
+		_color["prim"] = href_list["icon"]
+		update_icon()

```

Comment:
You're right, I'll have to take a closer look at this later and add a safety check.